File: cpp_solutions/298. Binary Tree Longest Consecutive Sequence.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int max_h = 0;
    
    int find_ans(TreeNode* root, int h, int prev) {
        max_h = max(h, max_h);
        if (root == NULL) {
            return h;
        } 
        if (root->val == prev + 1) {
            return max(find_ans(root->right, h + 1, root->val), find_ans(root->left, h + 1, root->val));
        } else {
            return max(find_ans(root->right, 1, root->val), find_ans(root->left, 1, root->val));
        }
    }
    int longestConsecutive(TreeNode* root) {
        if (root == NULL) {
            return 0;
        }
        find_ans(root, 1, -100000);
        return max_h;
    }
};

File: python3_solution_set1/298. Binary Tree Longest Consecutive Sequence.py
##ss
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def longestConsecutive(self, root: Optional[TreeNode]) -> int:
        
        self.ans = 0
        self.makeseq(root,root.val-1,0)
        return self.ans
    def makeseq(self,root,prev,count):
        #print(count,root.val)
        self.ans = max(self.ans,count)
        if root is not None and root.val == prev+1:
            
            if root.left:
                self.makeseq(root.left,root.val,count+1)
                
            if root.right:
                self.makeseq(root.right,root.val,count+1)
                
            self.ans = max(self.ans,count+1)
                
        elif root is not None:
            
            if root.left:
                self.makeseq(root.left,root.val,1)
                
            if root.right:
                self.makeseq(root.right,root.val,1)
                
        
